atlas-db-controller/controller.go

```diff
@@ -581,21 +581,21 @@ func (c *Controller) syncDatabase(key string) error {
 		}
 	}
 
-	err = p.SyncDatabase(db, dsn)
+	// Update dsn related to a database which databaseschema will use.
+	err = c.syncDatabaseSecret(key, db, s)
 	if err != nil {
-		msg := fmt.Sprintf("error syncing database '%s': %s", key, err)
+		msg := fmt.Sprintf("error syncing database secrets '%s': %s", key, err)
 		c.updateDatabaseStatus(key, db, StateError, msg)
 		runtime.HandleError(fmt.Errorf(msg))
+		// TODO: question should return error or nil.
```

Comment:
What does the caller do with it? Is it sufficient to have updated the status? If the caller is going to ignore this error then you can return nil (in general I believe if we call runtime.HandleError it is because we are in a context that cannot return the error, or at least that the error is fully handled already).

Reply:
we think it is sufficient to have updated the status.  So will do "return nil"